agent/engine/docker_container_engine.go

```diff
@@ -573,6 +571,12 @@ func (dg *DockerGoClient) ContainerEvents(ctx context.Context) (<-chan DockerCon
 				// These result in us falling through to inspect the container, some
 				// out of caution, some because it's a form of state change
 
+			case "oom":
+				seelog.Infof("process within container %v died due to OOM", event.ID)
```

Comment:
Agreed that would be awesome, but that's too big for this PR; it requires a fairly significant amount of work to get that. We probably want the same behavior for resize and rename (if you do it to a container we're managing, log a warning with the task info).